privacyidea/lib/tokens/tantoken.py

```diff
@@ -126,9 +126,15 @@ def update(self, param, reset_failcount=True):
             # init tokens with tans
             tans = param.get("tans").split()
             tan_dict = {k: v for k, v in enumerate(tans)}
+            # Avoid to generate TANs, since we get the from params
+            param["tantoken_count"] = 0
+            # Determine the otplen from the TANs
+            if len(tans) > 0:
+                param["otplen"] = len(tans[0])
+            PaperTokenClass.update(self, param, reset_failcount=reset_failcount)
         else:
             # Init token without tans, so we create tans
-            param["papertoken_count"] = param.get("tantoken_count") or DEFAULT_COUNT
+            param["tantoken_count"] = param.get("tantoken_count") or DEFAULT_COUNT
```

Comment:
I think if we rename the parameter here, the underlying ``PaperTokenClass.update`` will always generate 100 TANs because it gets the number of OTPs to generate from the non-existing ``papertoken_count`` parameter.

Reply:
Oh, you are right. Now I know, why the parameter was used this way...
I will rechange this!